AppInspector/MetaDataHelper.cs

```diff
@@ -130,7 +123,10 @@ public void AddMatchRecord(MatchRecord matchRecord)
                     _ = UniqueTags.TryAdd(tag,0);
                 }
 
-                Metadata.Matches.Add(matchRecord);
+                if (allowAdd)
+                {
+                    Metadata.Matches.Add(matchRecord);
```

Comment:
I recommend that Metadata be an instance class held and returned by metadatahelper rather than a static class.  Metadata ultimately is holding instance data - not static data - so it should not be static.

Reply:
The current declaration in MetaDataHelper is:   internal MetaData Metadata { get; set; } so it is an instance already not static.  Note because it takes an arg in the constructor it is not initialized inline.  Anyway, this is why I'm a bit confused.  It makes it's internal instance available to the caller which is used by AnalyzeCommand as the final result in GetResult();  I know I'm missing something in your suggestion.  Thanks.

Reply:
Thanks. I thought it was static. My misunderstanding.